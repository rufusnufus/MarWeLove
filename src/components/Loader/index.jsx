import React, { useRef, useEffect } from 'react'
import useOnScreen from '../../hooks/useOnScreen'

function Loader({ onChange }) {
  const ref = useRef()
  const isVisible = useOnScreen(ref)

  useEffect(() => {
    if (isVisible) {
      onChange()
    }
  }, [isVisible])

  return <img ref={ref} className="loading" src="/images/loading.gif" alt="loading" />
}

export default Loader
