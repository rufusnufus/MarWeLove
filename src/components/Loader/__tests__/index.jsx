/**
 * @jest-environment jsdom
 */

import React from 'react'
import { cleanup, render } from '@testing-library/react'
import Loader from '../index'

jest.mock('../../../hooks/useOnScreen', () => () => true)
afterEach(cleanup)

describe('Loader', () => {
  it('Render on screen', () => {
    const { container } = render(<Loader onChange={jest.fn()} />)

    expect(container).toMatchSnapshot()
  })
})
