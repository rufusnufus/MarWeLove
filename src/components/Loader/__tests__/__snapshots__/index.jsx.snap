// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Render on screen 1`] = `
<div>
  <img
    alt="loading"
    class="loading"
    src="/images/loading.gif"
  />
</div>
`;
